### PR TITLE
allow custom source url

### PR DESCRIPTION
**Pull request by steve02081504**

### CODE REVIEW

File: public/scripts/char-data.js [hunk 1]
```diff
@@ -70,6 +70,8 @@
  * @property {"system" | "user" | "assistant"} depth_prompt.role - The role the character takes on during the prompted interaction (system, user, or assistant).
  * // Non-standard extensions added by external tools
  * @property {string} [pygmalion_id] - The unique identifier assigned to the character by the Pygmalion.chat.
+ * @property {`${string}/${string}`} [github_repo] - The gitHub repository associated with the character.
```
Comment (Cohee1207) [inline public/scripts/char-data.js]: I'm not sure that's a valid type. Can you just leave it as a string? In the end, it resolves to `string/string` being the only allowed value.

File: public/script.js [hunk 2]
```diff
@@ -1524,6 +1524,18 @@ function getCharacterSource(chId = this_chid) {
         return `https://pygmalion.chat/${pygmalionId}`;
     }
 
+    const githubRepo = characters[chId]?.data?.extensions?.github_repo;
+
+    if (githubRepo) {
+        return `https://github.com/${githubRepo}`;
+    }
+
+    const SourceUrl = characters[chId]?.data?.extensions?.source_url;
```
Comment (Cohee1207) [inline public/script.js]: consider proper `camelCasing` for local variables.

Comment (steve02081504) [reply public/script.js]: fixed in [`cd95028` (#2255)](https://github.com/SillyTavern/SillyTavern/pull/2255/commits/cd950287a3617363985d08152a0a8d668f0f1db5)

File: public/scripts/char-data.js (hunk 1 above)
Comment (steve02081504) [reply public/scripts/char-data.js]: fixed in [`cd95028` (#2255)](https://github.com/SillyTavern/SillyTavern/pull/2255/commits/cd950287a3617363985d08152a0a8d668f0f1db5)

**Pull request merged**